src/main/java/gyro/google/compute/DiskFinder.java

```diff
@@ -0,0 +1,120 @@
+import com.google.api.services.compute.model.DiskAggregatedList;
+import com.google.api.services.compute.model.DisksScopedList;
+import gyro.core.GyroException;
+import gyro.core.Type;
+import gyro.google.GoogleFinder;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Query for a zonal disk.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    compute-disk: $(external-query google::compute-disk { name: 'disk-example', zone: 'us-east1-b' })
+ */
+@Type("compute-disk")
+public class DiskFinder extends GoogleFinder<Compute, Disk, DiskResource> {
+    private String name;
+    private String zone;
+
+    /**
+     * The name of the disk.
+     */
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * The zone of the disk.
+     */
+    public String getZone() {
+        return zone;
+    }
+
+    public void setZone(String zone) {
+        this.zone = zone;
+    }
+
+    @Override
+    protected List<Disk> findAllGoogle(Compute client) {
+        try {
+            List<Disk> disks = new ArrayList<>();
+            DiskAggregatedList diskAggregatedList;
+            String nextPageToken = null;
+
+            do {
+                diskAggregatedList = client.disks().aggregatedList(getProjectId()).setPageToken(nextPageToken).execute();
+                disks.addAll(diskAggregatedList
+                    .getItems().values().stream()
+                    .map(DisksScopedList::getDisks)
+                    .filter(Objects::nonNull)
+                    .flatMap(Collection::stream)
+                    .filter(disk -> disk.getZone() != null)
+                    .collect(Collectors.toList()));
+                nextPageToken = diskAggregatedList.getNextPageToken();
+            } while (nextPageToken != null);
+
+            return disks;
+        } catch (GoogleJsonResponseException je) {
+            throw new GyroException(je.getDetails().getMessage());
+        } catch (IOException ex) {
+            throw new GyroException(ex);
+        }
+    }
+
+    @Override
+    protected List<Disk> findGoogle(Compute client, Map<String, String> filters) {
+        try {
+            if (filters.containsKey("name")) {
+                return Collections.singletonList(client.disks().get(getProjectId(), filters.get("zone"), filters.get("name")).execute());
+            } else {
+                return Optional.ofNullable(client.disks().list(getProjectId(), filters.get("zone")).execute().getItems())
+                    .orElse(new ArrayList<>());
+            }
+        } catch (GoogleJsonResponseException je) {
+            if (!je.getDetails().getMessage().matches("The resource (.*) was not found")) {
```

Comment:
Better to check with the error code than the message.

```suggestion
            if (je.getDetails().getCode() != 404) {
```